MiniApp/Classes/MiniAppDownloader.swift

```diff
@@ -23,27 +23,56 @@ class MiniAppDownloader {
 
     func download(appId: String, versionId: String, completionHandler: @escaping (Result<URL, Error>) -> Void) {
         let miniAppStoragePath = FileManager.getMiniAppVersionDirectory(with: appId, and: versionId)
-        if miniAppStatus.isDownloaded(appId: appId, versionId: versionId) {
+        if !isMiniAppAlreadyDownloaded(appId: appId, versionId: versionId) {
+            self.manifestDownloader.fetchManifest(apiClient: self.miniAppClient, appId: appId, versionId: versionId) { (result) in
+                 switch result {
+                 case .success(let responseData):
+                     self.downloadMiniApp(urls: responseData.manifest, to: miniAppStoragePath) { downloadResult in
+                         switch downloadResult {
+                         case .success:
+                             self.miniAppStorage.cleanVersions(for: appId, differentFrom: versionId, status: self.miniAppStatus)
+
+                             fallthrough
+                         default:
+                             completionHandler(downloadResult)
+                         }
+                     }
+                 case .failure(let error):
+                     let downloadError = error as NSError
+                     if downloadError.code == NSURLErrorNotConnectedToInternet {
```

Comment:
In this case (when it fails to fetch the manifest for a mini app), shouldn't it always try to open the cached Mini App version regardless of the type of error, not just when it's `NSURLErrorNotConnectedToInternet? For example, the device could have temporary connection issues that cause the request to time out. I think this is the approach that Android took.

Reply:
Ah sorry, I just checked Android and it's only catching network errors: `UnknownHostException` and `SocketTimeoutException`. So is there an Error type for "timeout" which you can check here? And any other network error types that should be checked?

Reply:
Done 👍 . Added Timeout exception as well